So, yeah, still the same problem as #51217:

```
2025-11-26T09:34:52.1106062Z 2025-11-26 09:34:52,005 ip-10-1-45-240 /home/runner/_work/quarkus/quarkus/integration-tests/jpa-postgresql/target/quarkus-integration-test-jpa-postgresql-999-SNAPSHOT-runner[2924] ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /jpa/testfunctionality/json failed, error id: 43af0c07-aebe-4218-b897-23156b9bcfc8-1: java.lang.IllegalArgumentException: Could not serialize object of java type: class io.quarkus.it.jpa.postgresql.defaultpu.SomeEmbeddable$ToBeSerializedWithDateTime
2025-11-26T09:34:52.1109204Z 	at org.hibernate.type.format.jackson.JacksonJsonFormatMapper.toString(JacksonJsonFormatMapper.java:86)
2025-11-26T09:34:52.1110373Z 	at org.hibernate.type.format.AbstractJsonFormatMapper.toString(AbstractJsonFormatMapper.java:33)
2025-11-26T09:34:52.1111701Z 	at org.hibernate.type.descriptor.java.spi.FormatMapperBasedJavaType.toString(FormatMapperBasedJavaType.java:56)
2025-11-26T09:34:52.1113021Z 	at org.hibernate.type.descriptor.java.spi.FormatMapperBasedJavaType.deepCopy(FormatMapperBasedJavaType.java:109)
2025-11-26T09:34:52.1114520Z 	at org.hibernate.type.AbstractStandardBasicType.deepCopy(AbstractStandardBasicType.java:280)
2025-11-26T09:34:52.1115714Z 	at org.hibernate.type.AbstractStandardBasicType.deepCopy(AbstractStandardBasicType.java:276)
2025-11-26T09:34:52.1116576Z 	at org.hibernate.type.ComponentType.deepCopy(ComponentType.java:509)
2025-11-26T09:34:52.1117407Z 	at org.hibernate.collection.spi.PersistentSet.getSnapshot(PersistentSet.java:80)
2025-11-26T09:34:52.1118320Z 	at org.hibernate.engine.spi.CollectionEntry.<init>(CollectionEntry.java:75)
2025-11-26T09:34:52.1119361Z 	at org.hibernate.engine.internal.StatefulPersistenceContext.addCollection(StatefulPersistenceContext.java:1096)
2025-11-26T09:34:52.1120625Z 	at org.hibernate.engine.internal.StatefulPersistenceContext.addNewCollection(StatefulPersistenceContext.java:1028)
2025-11-26T09:34:52.1121922Z 	at org.hibernate.event.internal.WrapVisitor.processArrayOrNewCollection(WrapVisitor.java:126)
2025-11-26T09:34:52.1122962Z 	at org.hibernate.event.internal.WrapVisitor.processCollection(WrapVisitor.java:66)
2025-11-26T09:34:52.1124422Z 	at org.hibernate.event.internal.AbstractVisitor.processValue(AbstractVisitor.java:88)
2025-11-26T09:34:52.1125295Z 	at org.hibernate.event.internal.WrapVisitor.processValue(WrapVisitor.java:137)
2025-11-26T09:34:52.1126546Z 	at org.hibernate.event.internal.AbstractVisitor.processEntityPropertyValues(AbstractVisitor.java:53)
2025-11-26T09:34:52.1127819Z 	at org.hibernate.event.internal.AbstractSaveEventListener.visitCollectionsBeforeSave(AbstractSaveEventListener.java:417)
2025-11-26T09:34:52.1129307Z 	at org.hibernate.event.internal.AbstractSaveEventListener.cloneAndSubstituteValues(AbstractSaveEventListener.java:342)
2025-11-26T09:34:52.1130612Z 	at org.hibernate.event.internal.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:291)
2025-11-26T09:34:52.1135981Z 	at org.hibernate.event.internal.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:220)
2025-11-26T09:34:52.1137243Z 	at org.hibernate.event.internal.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:138)
2025-11-26T09:34:52.1138564Z 	at org.hibernate.event.internal.DefaultPersistEventListener.entityIsTransient(DefaultPersistEventListener.java:148)
2025-11-26T09:34:52.1139861Z 	at org.hibernate.event.internal.DefaultPersistEventListener.persist(DefaultPersistEventListener.java:92)
2025-11-26T09:34:52.1141097Z 	at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:76)
2025-11-26T09:34:52.1142301Z 	at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:52)
2025-11-26T09:34:52.1143810Z 	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:140)
2025-11-26T09:34:52.1144881Z 	at org.hibernate.internal.SessionImpl.firePersist(SessionImpl.java:692)
2025-11-26T09:34:52.1145685Z 	at org.hibernate.internal.SessionImpl.persist(SessionImpl.java:676)
2025-11-26T09:34:52.1146743Z 	at io.quarkus.hibernate.orm.runtime.session.TransactionScopedSession.persist(TransactionScopedSession.java:167)
2025-11-26T09:34:52.1147891Z 	at org.hibernate.engine.spi.SessionLazyDelegator.persist(SessionLazyDelegator.java:221)
2025-11-26T09:34:52.1148823Z 	at org.hibernate.Session_x03UFp930IxQOyJJLmtWw2baujI_Synthetic_ClientProxy.persist(Unknown Source)
2025-11-26T09:34:52.1150108Z 	at io.quarkus.it.jpa.postgresql.JPAFunctionalityTestEndpoint.lambda$json$7(JPAFunctionalityTestEndpoint.java:139)
2025-11-26T09:34:52.1151287Z 	at io.quarkus.narayana.jta.TransactionRunnerImpl.lambda$run$0(TransactionRunnerImpl.java:27)
2025-11-26T09:34:52.1152315Z 	at io.quarkus.narayana.jta.QuarkusTransactionImpl.callInOurTx(QuarkusTransactionImpl.java:136)
2025-11-26T09:34:52.1163440Z 	at io.quarkus.narayana.jta.QuarkusTransactionImpl.callRequireNew(QuarkusTransactionImpl.java:106)
2025-11-26T09:34:52.1164957Z 	at io.quarkus.narayana.jta.QuarkusTransactionImpl.call(QuarkusTransactionImpl.java:29)
2025-11-26T09:34:52.1166059Z 	at io.quarkus.narayana.jta.TransactionRunnerImpl.run(TransactionRunnerImpl.java:26)
2025-11-26T09:34:52.1167599Z 	at io.quarkus.it.jpa.postgresql.JPAFunctionalityTestEndpoint.json(JPAFunctionalityTestEndpoint.java:135)
2025-11-26T09:34:52.1169231Z 	at io.quarkus.it.jpa.postgresql.JPAFunctionalityTestEndpoint$quarkusrestinvoker$json_f715b9dedd84ff6d785ce1f3e42f9d5c556de8f4.invoke(Unknown Source)
2025-11-26T09:34:52.1171051Z 	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
2025-11-26T09:34:52.1172918Z 	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:183)
2025-11-26T09:34:52.1176178Z 	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
2025-11-26T09:34:52.1177960Z 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$15.runWith(VertxCoreRecorder.java:645)
2025-11-26T09:34:52.1179232Z 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
2025-11-26T09:34:52.1180358Z 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
2025-11-26T09:34:52.1181759Z 	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
2025-11-26T09:34:52.1183580Z 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
2025-11-26T09:34:52.1184625Z 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
2025-11-26T09:34:52.1185741Z 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
2025-11-26T09:34:52.1186768Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2025-11-26T09:34:52.1187677Z 	at java.base@21.0.9/java.lang.Thread.runWith(Thread.java:1596)
2025-11-26T09:34:52.1188365Z 	at java.base@21.0.9/java.lang.Thread.run(Thread.java:1583)
2025-11-26T09:34:52.1189444Z 	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)
2025-11-26T09:34:52.1190848Z 	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)
2025-11-26T09:34:52.1194040Z Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.LocalDate` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling (or disable `MapperFeature.REQUIRE_HANDLERS_FOR_JAVA8_TIMES`) (through reference chain: io.quarkus.it.jpa.postgresql.defaultpu.SomeEmbeddable$ToBeSerializedWithDateTime["date"])
2025-11-26T09:34:52.1197416Z 	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1359)
2025-11-26T09:34:52.1199992Z 	at com.fasterxml.jackson.databind.ser.impl.UnsupportedTypeSerializer.serialize(UnsupportedTypeSerializer.java:34)
2025-11-26T09:34:52.1203563Z 	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
2025-11-26T09:34:52.1205408Z 	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
2025-11-26T09:34:52.1206682Z 	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
2025-11-26T09:34:52.1208164Z 	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:503)
2025-11-26T09:34:52.1210321Z 	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:423)
2025-11-26T09:34:52.1211789Z 	at com.fasterxml.jackson.databind.ObjectWriter$Prefetch.serialize(ObjectWriter.java:1583)
2025-11-26T09:34:52.1213207Z 	at com.fasterxml.jackson.databind.ObjectWriter._writeValueAndClose(ObjectWriter.java:1289)
2025-11-26T09:34:52.1214522Z 	at com.fasterxml.jackson.databind.ObjectWriter.writeValueAsString(ObjectWriter.java:1140)
2025-11-26T09:34:52.1215840Z 	at org.hibernate.type.format.jackson.JacksonJsonFormatMapper.toString(JacksonJsonFormatMapper.java:83)
2025-11-26T09:34:52.1217100Z 	... 53 more
```

--- a/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
+++ b/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
@@ -63,6 +63,8 @@ import org.jboss.jandex.Indexer;
 import org.jboss.logging.Logger;
 import org.jboss.logmanager.Level;
 
+import com.fasterxml.jackson.databind.Module;
+
 import io.quarkus.agroal.spi.JdbcDataSourceBuildItem;
 import io.quarkus.agroal.spi.JdbcDataSourceSchemaReadyBuildItem;
 import io.quarkus.arc.deployment.BeanContainerBuildItem;
@@ -318,6 +320,19 @@ public final class HibernateOrmProcessor {
         } else {
             return ImpliedBlockingPersistenceUnitTypeBuildItem.generateImpliedPersistenceUnit();
         }
+    }
+
+    @BuildStep
+    public void allowJacksonModuleDiscovery(Capabilities capabilities,
+            List<PersistenceUnitDescriptorBuildItem> persistenceUnits,
+            BuildProducer<ServiceProviderBuildItem> serviceProviders) {
+        if (capabilities.isMissing(Capability.JACKSON) || persistenceUnits.isEmpty()) {
+            // We won't be using Hibernate's default FormatMapper relying on Jackson for sure
+            return;
+        }
+        // Hibernate's default FormatMapper relying on Jackson requires
+        // service loading to discover modules in the classpath.
+        serviceProviders.produce(ServiceProviderBuildItem.allProvidersFromClassPath(Module.class.getName()));
     }
 
     @BuildStep

--- a/pom.xml
+++ b/pom.xml
@@ -71,12 +71,12 @@
         <jacoco.version>0.8.14</jacoco.version>
         <kubernetes-client.version>7.4.0</kubernetes-client.version> <!-- Please check with Java Operator SDK team before updating -->
         <rest-assured.version>5.5.6</rest-assured.version>
-        <hibernate-orm.version>7.1.6.Final</hibernate-orm.version> <!-- WARNING when updating, also align the versions below -->
+        <hibernate-orm.version>7.1.10.Final</hibernate-orm.version> <!-- WARNING when updating, also align the versions below -->
         <jakarta.persistence-api.version>3.2.0</jakarta.persistence-api.version> <!-- version controlled by Hibernate ORM's needs -->
         <antlr.version>4.13.2</antlr.version> <!-- version controlled by Hibernate ORM's needs -->
         <bytebuddy.version>1.17.6</bytebuddy.version> <!-- version controlled by Hibernate ORM's needs -->
         <hibernate-models.version>1.0.1</hibernate-models.version> <!-- version controlled by Hibernate ORM's needs -->
-        <hibernate-reactive.version>3.1.8.Final</hibernate-reactive.version> <!-- highly sensitive to Hibernate ORM upgrades -->
+        <hibernate-reactive.version>3.1.9.Final</hibernate-reactive.version> <!-- highly sensitive to Hibernate ORM upgrades -->
         <hibernate-validator.version>9.1.0.Final</hibernate-validator.version>
         <hibernate-search.version>8.1.2.Final</hibernate-search.version>
         <hibernate-tools.version>7.1.6.Final</hibernate-tools.version>

